core/domain/customization_args_util.py

```diff
@@ -91,33 +90,31 @@ def validate_customization_args_and_values(
             'Expected customization args to be a dict, received %s'
             % customization_args)
 
-    # Validate and clean up the customization args.
-
-    # Populate missing keys with the default values.
-    customization_args = get_full_customization_args(
-        customization_args, ca_specs_to_validate_against)
-
-    # Remove extra keys.
-    extra_args = []
+    # Check for extra invalid keys.
     for arg_name in customization_args.keys():
         if not isinstance(arg_name, python_utils.BASESTRING):
             raise utils.ValidationError(
                 'Invalid customization arg name: %s' % arg_name)
         if arg_name not in ca_spec_names:
-            extra_args.append(arg_name)
-            logging.warning(
+            raise utils.ValidationError(
                 '%s %s does not support customization arg %s.'
                 % (item_name.capitalize(), item_type, arg_name))
-    for extra_arg in extra_args:
-        del customization_args[extra_arg]
 
     # Check that each value has the correct type.
     for ca_spec in ca_specs_to_validate_against:
+        if ca_spec.name not in customization_args:
+            raise utils.ValidationError(
+                'Customization argument is missing key: %s' % ca_spec.name)
         try:
+            # Since SubtitledHtml is a custom object in objects.py,
```

Comment:
Updated to:
```
# Since SubtitledHtml is a custom object which contains a html
            # field as one of the properties in its dict schema, we use
            # apply_custom_validators=False so that normalize_against_schema
            # traverses to the nested html schema inside SubtitledHtml and
            # properly normzalizes the html.
            # Using apply_custom_validators=True here would apply the
            # SubtitledHtml's custom validator in objects.py (which is not
            # implemented) and result in normalize_against_schema
            # not traversing to the nested html schema.
```
Hopefully that is more clear